src/main/java/org/folio/reader/JPathSyntaxEntityReader.java

```diff
@@ -101,7 +102,7 @@ protected RuleValue readCompositeValue(Rule rule) {
       return MissingValue.getInstance();
     } else {
       CompositeValue compositeValue = buildCompositeValue(matrix);
-      applyReadDependingOnDataSourceFlag(compositeValue);
```

Comment:
From discussion with Igor Gorchakov, this part of the code was responsible only for managing holding hr id subfield and omitting the item fields in case of null values.